File: go/2220.go
package leetcode

import (
	"strconv"
	"strings"
)

func minBitFlips(start int, goal int) int {
	binS := strconv.FormatInt(int64(start), 2)
	binG := strconv.FormatInt(int64(goal), 2)
	maxLen := max(len(binS), len(binG))
	s := strings.Repeat("0", maxLen-len(binS)) + binS
	g := strings.Repeat("0", maxLen-len(binG)) + binG
	ans := 0
	for i := 0; i < maxLen; i++ {
		if s[i] != g[i] {
			ans += 1
		}
	}
	return ans
}

File: javascript/2220.js
/**
 * @param {number} start
 * @param {number} goal
 * @return {number}
 */
var minBitFlips = function (start, goal) {
  let binS = start.toString(2);
  let binG = goal.toString(2);
  const maxLen = Math.max(binS.length, binG.length);
  let s = "0".repeat(maxLen - binS.length) + binS;
  let g = "0".repeat(maxLen - binG.length) + binG;
  let ans = 0;
  for (let i = 0; i < maxLen; i++) {
    if (s[i] != g[i]) {
      ans += 1;
    }
  }
  return ans;
};
